Protocols/Http/Request.php

```diff
@@ -458,13 +458,17 @@ protected function parsePost()
             $this->_data['post'] = static::$_postCache[$body_buffer];
             return;
         }
-        $content_type = $this->header('content-type');
-        if ($content_type !== null && \preg_match('/boundary="?(\S+)"?/', $content_type, $match)) {
+        $content_type = $this->header('content-type', '');
+        if (\preg_match('/boundary="?(\S+)"?/', $content_type, $match)) {
             $http_post_boundary = '--' . $match[1];
             $this->parseUploadFiles($http_post_boundary);
             return;
         }
-        \parse_str($body_buffer, $this->_data['post']);
+        if (\preg_match('/application\/json/i', $content_type)) {
```

Comment:
I propose to match `json` like this:
https://github.com/symfony/symfony/blob/dc6e8d0ffc4cdbde7bfca886fafb38d4f4cfb308/src/Symfony/Component/HttpClient/Response/CommonResponseTrait.php#L93

Reply:
Thanks! I believe Symfony's codes have been well tested for a long time so I am convinced that's safe.

Reply:
`\b` just matches a word boundary (non-`\w`). It's used to match some non-standard types like `application/x-amz-json-1.1`